### PR TITLE
[MAINT] Disable ruff rules conflicting with the formatter

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Also disable deprecated and problematic rules.

Addresses part of #4739.

### CODE REVIEW

**Review by bthirion — COMMENTED** [review]
No opinion on this.

File: pyproject.toml [hunk 1]
```diff
@@ -150,7 +150,24 @@ ignore = [
...
+    "D206",
+    "D300",
+    "Q000",
+    "Q001",
+    "Q002",
+    "Q003",
+    "COM812",
+    "COM819",
+    "ISC001",
+    "ISC002"
```
Comment (Remi-Gau) [inline pyproject.toml:170]: OK with me but can I ask you to sort those alpha-numerically for the sake of our common long term mental heatlh. 🙏🏽

**Review by Remi-Gau — CHANGES_REQUESTED** [review]
one nitpick

File: pyproject.toml (hunk 1 above)
Comment (DimitriPapadopoulos) [reply pyproject.toml:170]: I kept the order of the documentation. That's something I can change of course. Is it OK to keep the conflicting rules as a separate section?

```
    .
    .
    .
    "UP027",  # deprecated
    "UP038",  # https://github.com/astral-sh/ruff/issues/7871
    # https://docs.astral.sh/ruff/formatter/#conflicting-lint-rules
    "COM812",
    "COM819",
    .
    .
    .
    "W191"
```

Comment (Remi-Gau) [reply pyproject.toml:170]: YES I was thinking to actually keep that comment as a separator so we know which ones come from us or not: don't hesitate to make that comment even more clear about this !!!

Comment (DimitriPapadopoulos) [reply pyproject.toml]: Done.

**Comment by DimitriPapadopoulos** [comment]
While we're at it, I would suggest changing [`select`](https://docs.astral.sh/ruff/settings/#lint_select) to [`extend-select`](https://docs.astral.sh/ruff/settings/#lint_extend-select):
https://github.com/nilearn/nilearn/blob/d0231a4446a2cad4811987fc8750b93972551041/pyproject.toml#L156-L158

This ensures **default rules** are selected as ruff matures.

**Review by Remi-Gau — APPROVED** [review]
thanks !!

**Pull request merged**